sw/device/tests/sim_dv/inject_scramble_seed.c

```diff
@@ -40,7 +40,7 @@ OTTF_DEFINE_TEST_CONFIG();
 // test environment.  The data has to be backdoor loaded because the
 // life cycle state used in this test does not permit the isolated
 // partition to be written.
-static volatile const uint32_t kIsoPartExpData[16];
+OT_SECTION(".non_volatile_scratch") uint32_t kIsoPartExpData[16];
```

Comment:
Can you explain why 'static volatile const' didn't work?
I have non-flash test using 'static volatile const' wondering if 
I have to make the same update.

Reply:
ah this is because the test itself is kind of stupid. 
`kIsoPartExpData` is backdoor loaded by the test sequence in "unscrambled" form. 

So that means, wherever that data is, i need to set scramble_en to 0 in order to properly read that back.
The problem is, all other data loaded by the test itself is scrambled. So when I set scramble_en to 0, all the data except for `kIsoPartExpData` read back incorrectly. This caused the register program interface to program incorrect values and eventually break the test.

Reply:
So, the alternative would be to backdoor write the unscrambled data in Iso partition of the flash, and the corresponding scrambled data in the `.data` section for the SW to be able to read back?